sources/packages/web/src/views/institution/locations/active-applications/ActiveApplicationSummaryData.vue

```diff
@@ -0,0 +1,228 @@
+import { useFormatters } from "@/composables";
+import StatusChipActiveApplication from "@/components/generic/StatusChipActiveApplication.vue";
+const DEFAULT_SORT_FIELD = "applicationNumber";
+
+export default {
+  components: {
+    StatusChipActiveApplication,
+  },
+  props: {
+    locationId: {
+      type: Number,
+      required: true,
+    },
+    locationName: {
+      type: String,
+      required: true,
+    },
+    archived: {
+      type: Boolean,
+      required: true,
+    },
+  },
+
+  setup(props: any) {
+    const router = useRouter();
+    const page = ref(DEFAULT_PAGE_NUMBER);
+    const pageLimit = ref(DEFAULT_PAGE_LIMIT);
+    const sortField = ref(DEFAULT_SORT_FIELD);
+    const sortOrder = ref(DataTableSortOrder.ASC);
+    const searchCriteria = ref();
+    const { dateString } = useFormatters();
+    const applications = ref(
+      {} as PaginatedResults<ActiveApplicationSummaryAPIOutDTO>,
+    );
+    const rowsPerPageOptions = computed(() =>
+      applications.value.results?.length > 10 ? PAGINATION_LIST : undefined,
+    );
+
+    const goToViewApplication = (applicationId: number) => {
+      router.push({
+        name: InstitutionRoutesConst.ACTIVE_APPLICATION_EDIT,
+        params: { locationId: props.locationId, applicationId },
+      });
+    };
+
+    const goToViewScholasticStanding = (scholasticStandingId: number) => {
+      router.push({
+        name: InstitutionRoutesConst.SCHOLASTIC_STANDING_VIEW,
+        params: { locationId: props.locationId, scholasticStandingId },
+      });
+    };
+
+    const getSummaryList = async (locationId: number) => {
+      applications.value =
+        await InstitutionService.shared.getActiveApplicationsSummary(
+          locationId,
+          {
+            page: page.value,
+            pageLimit: pageLimit.value,
+            sortField: sortField.value,
+            sortOrder: sortOrder.value,
+            searchCriteria: searchCriteria.value,
+          },
+          props.archived,
+        );
+    };
+
+    const pageEvent = async (event: PageAndSortEvent) => {
+      page.value = event?.page;
+      pageLimit.value = event?.rows;
+      await getSummaryList(props.locationId);
+    };
+
+    const sortEvent = async (event: PageAndSortEvent) => {
+      page.value = DEFAULT_PAGE_NUMBER;
+      pageLimit.value = DEFAULT_PAGE_LIMIT;
+      sortField.value = event.sortField;
+      sortOrder.value = event.sortOrder;
+      await getSummaryList(props.locationId);
+    };
+
+    const searchActiveApplications = async () => {
+      page.value = DEFAULT_PAGE_NUMBER;
+      pageLimit.value = DEFAULT_PAGE_LIMIT;
+      await getSummaryList(props.locationId);
+    };
+
+    watch(
+      () => props.locationId,
+      async (currValue) => {
+        //update the list
+        await getSummaryList(currValue);
+      },
+    );
+
+    onMounted(async () => {
+      await getSummaryList(props.locationId);
```

Comment:
you can use ` { immediate: true },` in watch instead of `onMounted`. something like this 
![image](https://user-images.githubusercontent.com/77353155/174658685-6b36e25e-4fe7-4891-85c5-62c42059eac5.png)